### PR TITLE
Could not find publication: maven 

### PR DESCRIPTION
**Pull request by ouchadam**
**Problem**

Android releases were not working, failing with can't find `publication maven`

**Solution**

- removes default `maven` publication, we dynamically populate this value, but allow it to be overridden in the publish extension.
- hard codes the android `release` variant as the publication instead of allowing both `debug` and `release`
- removes applying the project name as the maven artifact id, that should be configured from outside the library if wanted.

Also ended up removing failing tests, we aren't able to maintain them since the original intention is lost 😢 and our CI environment does not support them.

paired with @frapontillo

### CODE REVIEW

File: core/src/gradleTest/java_lib/build.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.novoda:bintray-release:0.3.4'
+        classpath 'com.novoda:bintray-release:' + project.version
```
Comment (mr-archano) [inline core/src/gradleTest/java_lib/build.gradle]: string interpolation _bitte_ 🇩🇪  (idiomatic groovy and such)
```
classpath "com.novoda:bintray-release:$project.version"
```

File: core/src/main/groovy/com/novoda/gradle/release/BintrayConfiguration.groovy [hunk 2]
```diff
@@ -22,7 +22,7 @@ class BintrayConfiguration {
             publish = extension.autoPublish
             dryRun = propertyFinder.getDryRun()
 
-            publications = extension.publications ?: project.plugins.hasPlugin('com.android.library') ? project.android.libraryVariants.collect { it.name } : [ 'maven' ]
+            publications = extension.publications ?: project.plugins.findPlugin('com.android.library') ? ['release'] : [ 'maven' ]
```
Comment (mr-archano) [inline core/src/main/groovy/com/novoda/gradle/release/BintrayConfiguration.groovy]: why the change to `findPlugin()`?

File: core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy [hunk 3]
```diff
@@ -18,15 +18,10 @@ class ReleasePlugin implements Plugin<Project> {
     }
 
     void attachArtifacts(Project project) {
-        if (project.plugins.hasPlugin('com.android.library')) {
+        if (project.plugins.findPlugin('com.android.library')) {
```
Comment (mr-archano) [inline core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy]: same here: what's the rationale of the change?

File: core/src/test/groovy/com/novoda/gradle/release/LibProjectFactory.groovy [hunk 4]
```diff
@@ -1,53 +0,0 @@
-package com.novoda.gradle.release
-
-import org.gradle.api.Project
-import org.gradle.testfixtures.ProjectBuilder
-
-final class LibProjectFactory {
```
Comment (mr-archano) [inline core/src/test/groovy/com/novoda/gradle/release/LibProjectFactory.groovy]: I know a guy who created some pretty slick integration tests for Gradle plugins, you should ask him to look into this 👀

File: core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy (hunk 3 above)
Comment (ouchadam) [inline core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy]: ahhh we misread the docs, thought that hasPlugin was being removed, will revert

File: core/src/test/groovy/com/novoda/gradle/release/LibProjectFactory.groovy (hunk 4 above)
Comment (hhaouat) [inline core/src/test/groovy/com/novoda/gradle/release/LibProjectFactory.groovy]: @mr-archano Who is this guy? :)

**Comment by hhaouat** [comment]
For my "little" point of view, it's seems to be fine. 😸  👍 
But maybe @xrigau can as well take a look like this we can get this PR merge for today. 🎉 
Actually this new release of `bintray-release` can help me to resolve errors on my build. :trollface:

File: core/src/test/groovy/com/novoda/gradle/release/LibProjectFactory.groovy (hunk 4 above)
Comment (mr-archano) [inline core/src/test/groovy/com/novoda/gradle/release/LibProjectFactory.groovy]: `shame-less plug warning:` as part of the work for the new shiny `static-analysis-plugin` we introduced a (quite extensive) [set](https://github.com/novoda/spikes/blob/master/static-analysis-plugin/plugin/src/test/groovy/com/novoda/test/TestProject.groovy) of [utilities](https://github.com/novoda/spikes/blob/master/static-analysis-plugin/plugin/src/test/groovy/com/novoda/test/TestProjectRule.groovy) that can be used to create synthetic Gradle projects in integration tests for plugins. We can apply a similar pattern here and bring back the integration tests.

**Comment by xrigau** [comment]
💯  nice action!! 🎉

**Comment by xrigau** [comment]
TI🌮

File: core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy (hunk 3 above)
Comment (frapontillo) [inline core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy]: We totally over🍺d
